### ISSUE TITLE
Realtime Injector accept string value arguments

### ISSUE THREAD

**Issue by jasonlharvey** [nasa/trick#1817]
Referencing the RTI functions rti_add and rti_fire described in the wiki here: https://nasa.github.io/trick/documentation/simulation_capabilities/Realtime-Injector

I'd like to be able to send the value and thread_id to rti_add and rti_fire, respectively, as a string, so that they can be called from a VariableServer Python class.  I'm extending the VariableServer Python class in trick/share/trick/pymods/trick/variable_server.py to add calls to send values to the sim using the rti_add and rti_fire, instead of the default trick.add_var function, since RTI has more deterministic timing.  However, VariableServer can only send commands to the Trick input processor encoded as a string, but the rti_fire and rti_add functions have no overload to accept their value and thread_id as a string.  For example, i'm sending the command from VariableServer to Trick as 'trick.rti_add(the_answer, 42.0)' which causes an error like this:

```
    Traceback (most recent call last):
  File "<string>", line 1, in <module>
NotImplementedError: Wrong number or type of arguments for overloaded function 'RtiStager_Add'.
  Possible C/C++ prototypes are:
    Trick::RtiStager::Add(char *,double)
    Trick::RtiStager::Add(char *,long long)
```

A solution might be to add another overload of RtiStager:Add(char*, char*) and then convert the char * value to the correct data format of the variable, since Trick knows the destination variable's type (double, int, etc.), and throw an error if the string value can't be converted (badly formatted string, etc.)  This might be needed for both the rti_add function's value argument, and the rti_fire function's thread_id argument.

**Comment by hchen99** [comment]
Wondering if something like `trick.rti_add("test_so.obj.my_char", float("42.0"))` would work?

**Comment by jasonlharvey** [comment]
I tried this but it gives the same error.  I suspect that instead of sending a float data type to RtiStager::Add, it's still trying to send it as a string, now just "float(42.0)" instead of "42.0". 

Here is the python in my extended VariableServer class and the command string that is sent to the VariableServer's send function:

```
self.send('trick.rti_add(' + name + ', float(' + str(value) + '))')
```

In this context, name is a python string and value is a python float.

**Comment by alexlin0** [comment]
Should name be enclosed in quotes?  Is this a difference in sending

`trick.rti_add(var.name, 42.0)`
vs.
`trick.rti_add("var.name", 42.0)`

Should your send function call look like this with a couple of added double quotes?

`self.send('trick.rti_add("' + name + '", float(' + str(value) + '))')`

**Comment by alexlin0** [comment]
I don't think the float call is necessary

`self.send('trick.rti_add("' + name + '", ' + str(value) + ')')`

**Comment by jasonlharvey** [comment]
I tried both your suggestions Alex and they both work!

So I was sending 'trick.rti_add(myvariablename, 42.0)' when I needed to send 'trick_rti_add("myvariablename", 42.0)'

I guess since the extra float typecast isn't needed, i'll leave it out.  Similarly, for the thread_id given to the rti_fire command, it doesn't seem to need a typecast from the ID as a python string (i.e. '0') to int.  Both of these work:

```
self.send('trick.rti_fire(int(' + thread_id + '))')
```

and

```
self.send('trick.rti_fire(' + thread_id + ')')
```

Ok my stuff is working now, it was user error.  I withdraw the issue, thanks for the help!
